### PR TITLE
Store object_manifest for delivery

### LINKED ISSUE #151: record file metadata for s3 delivery
Record file metadata so we can verify that files are the same as those were delivered.

The s3 storage doesn't make a md5 sum available for every file upload. There is an etag field that stores md5 for small files, but for larger files a different value is stored.
The [boto docs]( http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Client.get_object) describe etag as
>ETag (string) -- An ETag is an opaque identifier assigned by a web server to a specific version of a resource found at a URL

Amazon encourages users to store the md5 in a metadata tag: https://aws.amazon.com/premiumsupport/knowledge-center/data-integrity-s3/

If we record the metadata for all files in the delivery we can use this data to verify the delivery is valid.

### PR DESCRIPTION
**Pull request by johnbradley**
Reads metadata and other fields from s3.
Stores this value as JSON in a new field in delivery when delivery is sent.
New field is named `object_manifest`.

Fixes #151

### CODE REVIEW

File: d4s2_api/models.py [hunk 1]
```diff
@@ -406,6 +406,7 @@ class S3Delivery(DeliveryBase):
     from_user = models.ForeignKey(S3User, related_name='sent_deliveries')
     to_user = models.ForeignKey(S3User, related_name='received_deliveries')
     transfer_id = models.UUIDField(default=uuid.uuid4)
+    object_manifest = models.TextField(help_text='JSON array of object metadata from bucket at time of sending bucket')
```
[the remark this answers is not in the document]
Comment (dleehr) [reply d4s2_api/models.py]: Another benefit to JSONField is avoiding ambiguity in the default value. The default is now `''`, which fails `json.loads()`.

File: d4s2_api_v2/api.py [hunk 2]
```diff
@@ -265,6 +266,19 @@ def _give_agent_permission(s3_delivery, user):
...
+    def _record_object_manifest(s3_delivery, user):
+        """
+        Update delivery recording the object manifest based on data in s3
+        :param s3_delivery: S3Delivery: details about what we will deliver
+        :param user: user who's credentials to use to talk to s3
+        """
+        bucket = s3_delivery.bucket
+        s3_bucket_util = S3BucketUtil(bucket.endpoint, user)
+        objects_manifest = s3_bucket_util.get_objects_manifest(bucket_name=bucket.name)
+        s3_delivery.object_manifest = json.dumps(objects_manifest)
```
Comment (dleehr) [inline d4s2_api_v2/api.py]: I think the responsibility of serializing this dictionary to/from JSON should be handled by the model. Another reason to use JSONField 😄

**Comment by johnbradley** [comment]
This now processes s3 delivery `send` in the background.

File: d4s2_api/models.py [hunk 3]
```diff
@@ -395,6 +396,10 @@ class Meta:
         unique_together = ('endpoint', 'name')
 
 
+class S3ObjectManifest(models.Model):
+    content = JSONField(help_text='JSON array of object metadata from bucket at time of sending bucket')
```
Comment (dleehr) [inline d4s2_api/models.py]: This is working nicely, I was able to write queries in python to walk through the internal JSON structure in postgres.

**Pull request merged**